### PR TITLE
[16.0][MIG] web_widget_dropdown_dynamic: Migrate to version 16.0

### PR DESCRIPTION
**Pull request by sonhd91**
Continuing with: 
* https://github.com/OCA/web/pull/2470

With the update to owl implementation, we have made some changes in the way we use this module in XML files

```
    <field name="other_field" />
    <field name="char_field" widget="dynamic_dropdown" options="{'values':'method_name'}" context="{'depending_on': other_field}" />
```

### CODE REVIEW

File: web_widget_dropdown_dynamic/static/src/js/basic_model.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019-2020 Brainbean Apps (https://brainbeanapps.com)
+ * License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+ */
+odoo.define("web_widget_dropdown_dynamic.basic_model", function (require) {
```
Comment (simahawk) [inline web_widget_dropdown_dynamic/static/src/js/basic_model.js]: any reason to keep this as old style JS?

**Review by simahawk — COMMENTED** [review]
LG overall

File: web_widget_dropdown_dynamic/static/src/js/basic_model.js (hunk 1 above)
Comment (sonhd91) [reply web_widget_dropdown_dynamic/static/src/js/basic_model.js]: I just updated it, please check again

**Review by simahawk — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]

**Comment by chienandalu** [comment]
This would be ready to merge, isn't it?

**Comment by pedrobaeza** [comment]
Can you please rebase and regenerate the README?

**Comment by sonhd91** [comment]
> Can you please rebase and regenerate the README?

Done

**Pull request merged**